### ISSUE TITLE
在检测自己的网络的过程中对weight的初始化出现了奇怪的问题

### ISSUE THREAD

**Issue by kanagi2587** [Oneflow-Inc/oneflow#3818]
weight_shape=(filters,inputs.shape[1]//groups,kernel_size_1,kernel_size_2,kernel_size_3)
    weight = flow.get_variable(
        name + "-weight",
        shape=weight_shape,
        dtype=inputs.dtype,
        initializer=weight_initializer,
        regularizer=weight_regularizer,
        trainable=trainable
    )
在某一层网络的构建过程中，出现1113 16:27:15.528882 16965 conv_op.cpp:81] Check failed: weight->shape() == Shape(weight_shape) ((128,128,1,3,3) vs. (128,256,1,3,3)) 
其中上述参数中filters为256，inputs.shape为(64,256,26,14,14)
对weight.shape和weight_shape进行打印发现输出分别为
(256, 256, 1, 3, 3)
(128, 128, 1, 3, 3)

**Comment by guo-ran** [comment]
请问调用conv时有没有传groups参数

**Comment by kanagi2587** [comment]
> 请问调用conv时有没有传groups参数？

一直使用的是某人groups=1，没有特别进行传参，而且这个用于构造conv3d的函数已经用了数次，这是在其中一次中出的问题

**Comment by kanagi2587** [comment]
> 请问调用conv时有没有传groups参数？

一直使用的是默认的groups=1，没有特别进行传参，而且这个用于构造conv3d的函数已经被调用了数次，这是在其中一次中出的问题

**Comment by guo-ran** [comment]
方便提供完整的脚本吗？或者包含input shape信息、及调用conv3d部分的部分脚本。

**Comment by kanagi2587** [comment]
> 方便提供完整的脚本吗？或者包含input shape信息、及调用conv3d部分的部分脚本。
 init=flow.random_uniform_initializer(minval=0, maxval=0.5)
out=conv3d_layer(
            "APP3DC_spatial",inputs,256 
            kernel_size=[1, 3,3], strides=[1, 2,2],
            padding=[0,0,0,1,1], use_bias=True,weight_initializer=init
        )
#inputs.shape=(64,256,26,14,14)
def conv3d_layer(
    name,
    inputs,
    filters,
    kernel_size=3,
    strides=1,
    padding="VALID",
    data_format="NCDHW",
    dilation_rate=1,
    activation="Relu",
    use_bias=False,
    groups=1,
    weight_initializer=_get_kernel_initializer(),
    bias_initializer=_get_bias_initializer(),
    weight_regularizer=_get_regularizer(),
    bias_regularizer=_get_regularizer(),
    trainable=True
):
    if isinstance(kernel_size,int):
        kernel_size_1=kernel_size
        kernel_size_2 = kernel_size
        kernel_size_3 = kernel_size
    if isinstance(kernel_size,list):
        kernel_size_1=kernel_size[0]
        kernel_size_2=kernel_size[1]
        kernel_size_3=kernel_size[2]
    weight_shape=(filters,inputs.shape[1]//groups,kernel_size_1,kernel_size_2,kernel_size_3)
    weight = flow.get_variable(
        name + "-weight",
        shape=weight_shape,
        dtype=inputs.dtype,
        initializer=weight_initializer,
        regularizer=weight_regularizer,
        trainable=trainable
    )
    output=flow.nn.conv3d(
         inputs, weight, strides, padding, data_format, dilation_rate, groups, name=name
    )
    if use_bias:
        bias = flow.get_variable(
            name + "-bias",
            shape=(filters,),
            dtype=inputs.dtype,
            initializer=bias_initializer,
            regularizer=bias_regularizer,
        )
        output = flow.nn.bias_add(output, bias, data_format)

    if activation is not None:
        if activation == "Relu":
            output = flow.nn.relu(output)
        else:
            raise NotImplementedError

    return output



在这一步之前有多次的类似的卷积等操作，但我觉得inputs的shape的不同应该不会导致这种问题

**Comment by guo-ran** [comment]
你好，我用你发的例子构建测试，没有复现你说的错误，测试文件代码如下：
```
import oneflow as flow
import numpy as np
import oneflow.typing as oft

# 随机模拟数据
func_config = flow.FunctionConfig()
func_config.default_data_type(flow.float)

x_shape = (64,256,26,14,14)
x = np.random.randn(*x_shape).astype(np.float32)

def _get_kernel_initializer():
    return flow.random_uniform_initializer(minval=0, maxval=0.5)

def _get_regularizer():
    return flow.regularizers.l2(0.0005)

def conv3d_layer(
        name,
        inputs,
        filters,
        kernel_size=3,
        strides=1,
        padding="VALID",
        data_format="NCDHW",
        dilation_rate=1,
        activation="Relu",
        use_bias=False,
        groups=1,
        weight_initializer=_get_kernel_initializer(),
        bias_initializer=_get_kernel_initializer(),
        weight_regularizer=_get_regularizer(),
        bias_regularizer=_get_regularizer(),
        trainable=True
    ):
    if isinstance(kernel_size,int):
        kernel_size_1=kernel_size
        kernel_size_2 = kernel_size
        kernel_size_3 = kernel_size
    if isinstance(kernel_size,list):
        kernel_size_1=kernel_size[0]
        kernel_size_2=kernel_size[1]
        kernel_size_3=kernel_size[2]
    weight_shape=(filters,inputs.shape[1]//groups,kernel_size_1,kernel_size_2,kernel_size_3)
    weight = flow.get_variable(
        name + "-weight",
        shape=weight_shape,
        dtype=inputs.dtype,
        initializer=weight_initializer,
        regularizer=weight_regularizer,
        trainable=trainable
        )
    output=flow.nn.conv3d(
        inputs, weight, strides, padding, data_format, dilation_rate, groups, name=name
    )
    if use_bias:
        bias = flow.get_variable(
        name + "-bias",
        shape=(filters,),
        dtype=inputs.dtype,
        initializer=bias_initializer,
        regularizer=bias_regularizer,
        )
    output = flow.nn.bias_add(output, bias, data_format)

    if activation is not None:
        if activation == "Relu":
            output = flow.nn.relu(output)
        else:
            raise NotImplementedError

    return output

@flow.global_function(type="train", function_config=func_config)
def FlowJob(x: oft.Numpy.Placeholder(x_shape, dtype=flow.float)):
    y=conv3d_layer(
        "APP3DC_spatial",x , 256,
        kernel_size=[1, 3, 3], strides=[1, 2, 2],
        padding=[0,0,0,1,1], use_bias=True, weight_initializer=flow.random_uniform_initializer(minval=0, maxval=0.5))
    
    flow.optimizer.SGD(
        flow.optimizer.PiecewiseConstantScheduler([], [1e-4]), momentum=0
    ).minimize(y)
    return y

check_point = flow.train.CheckPoint()
check_point.init()

for step in range(20):
    my_loss = FlowJob(x).get()
    print("my_loss", my_loss.numpy().mean())

**Comment by kanagi2587** [comment]
> 你好，我用你发的例子构建测试，没有复现你说的错误，测试文件代码如下：
> 
> ```
> import oneflow as flow
> import numpy as np
> import oneflow.typing as oft
> 
> # 随机模拟数据
> func_config = flow.FunctionConfig()
> func_config.default_data_type(flow.float)
> 
> x_shape = (64,256,26,14,14)
> x = np.random.randn(*x_shape).astype(np.float32)
> 
> def _get_kernel_initializer():
>     return flow.random_uniform_initializer(minval=0, maxval=0.5)
> 
> def _get_regularizer():
>     return flow.regularizers.l2(0.0005)
> 
> def conv3d_layer(
>         name,
>         inputs,
>         filters,
>         kernel_size=3,
>         strides=1,
>         padding="VALID",
>         data_format="NCDHW",
>         dilation_rate=1,
>         activation="Relu",
>         use_bias=False,
>         groups=1,
>         weight_initializer=_get_kernel_initializer(),
>         bias_initializer=_get_kernel_initializer(),
>         weight_regularizer=_get_regularizer(),
>         bias_regularizer=_get_regularizer(),
>         trainable=True
>     ):
>     if isinstance(kernel_size,int):
>         kernel_size_1=kernel_size
>         kernel_size_2 = kernel_size
>         kernel_size_3 = kernel_size
>     if isinstance(kernel_size,list):
>         kernel_size_1=kernel_size[0]
>         kernel_size_2=kernel_size[1]
>         kernel_size_3=kernel_size[2]
>     weight_shape=(filters,inputs.shape[1]//groups,kernel_size_1,kernel_size_2,kernel_size_3)
>     weight = flow.get_variable(
>         name + "-weight",
>         shape=weight_shape,
>         dtype=inputs.dtype,
>         initializer=weight_initializer,
>         regularizer=weight_regularizer,
>         trainable=trainable
>         )
>     output=flow.nn.conv3d(
>         inputs, weight, strides, padding, data_format, dilation_rate, groups, name=name
>     )
>     if use_bias:
>         bias = flow.get_variable(
>         name + "-bias",
>         shape=(filters,),
>         dtype=inputs.dtype,
>         initializer=bias_initializer,
>         regularizer=bias_regularizer,
>         )
>     output = flow.nn.bias_add(output, bias, data_format)
> 
>     if activation is not None:
>         if activation == "Relu":
>             output = flow.nn.relu(output)
>         else:
>             raise NotImplementedError
> 
>     return output
> 
> @flow.global_function(type="train", function_config=func_config)
> def FlowJob(x: oft.Numpy.Placeholder(x_shape, dtype=flow.float)):
>     y=conv3d_layer(
>         "APP3DC_spatial",x , 256,
>         kernel_size=[1, 3, 3], strides=[1, 2, 2],
>         padding=[0,0,0,1,1], use_bias=True, weight_initializer=flow.random_uniform_initializer(minval=0, maxval=0.5))
>     
>     flow.optimizer.SGD(
>         flow.optimizer.PiecewiseConstantScheduler([], [1e-4]), momentum=0
>     ).minimize(y)
>     return y
> 
> check_point = flow.train.CheckPoint()
> check_point.init()
> 
> for step in range(20):
>     my_loss = FlowJob(x).get()
>     print("my_loss", my_loss.numpy().mean())
>     
> ```

这也是我好奇问题出现的原因，因为之前的应用过程中也基本是正常的，目前的全部代码放在了https://github.com/kanagi2587/test/tree/master/random/models
运行testResnet.py文件在某一层出现了这个问题。

**Comment by guo-ran** [comment]
是因为网络中有两个以上variable定义成了相同的名字，导致了reuse，”varaible 'APP3DC_spatial_-weight' already exists“ 可以在get_variable函数加reuse=False, 参数禁用reuse帮助调试。

**Comment by kanagi2587** [comment]
> 是因为网络中有两个以上variable定义成了相同的名字，导致了reuse，”varaible 'APP3DC_spatial_-weight' already exists“ 可以在get_variable函数加reuse=False, 参数禁用reuse帮助调试。

感谢 是我疏忽了 仅仅修改了网络的重名 忽略了weight的

**Comment by guo-ran** [comment]
感谢，是我们的提示信息不够完善，我们会增加更多提示信息。
